Applications/spire/source/time_and_sales/snapshot_loading_widget.cpp

```diff
@@ -0,0 +1,52 @@
+#include "spire/time_and_sales/snapshot_loading_widget.hpp"
+#include <QMovie>
+#include <QResizeEvent>
+#include <QScrollBar>
+#include "spire/spire/dimensions.hpp"
+
+using namespace spire;
+
+snapshot_loading_widget::snapshot_loading_widget(QScrollArea* scroll_area,
+    QWidget* parent)
+    : QWidget(parent),
+      m_scroll_area(scroll_area) {
+  setAttribute(Qt::WA_TransparentForMouseEvents);
+  setFixedHeight(scale_height(32));
+  m_logo_widget = new QLabel(this);
+  auto logo = new QMovie(":/icons/pre-loader.gif", QByteArray(),
+    m_logo_widget);
+  logo->setScaledSize(scale(16, 16));
+  m_logo_widget->setMovie(logo);
+  m_logo_widget->setAlignment(Qt::AlignCenter);
+  m_logo_widget->movie()->start();
+  m_scroll_area->installEventFilter(this);
+  m_scroll_area->widget()->installEventFilter(this);
+  connect(m_scroll_area->horizontalScrollBar(), &QScrollBar::valueChanged, this,
+    [=] (auto i) { 
+      auto rect = m_scroll_area->widget()->visibleRegion().boundingRect();
```

Comment:
Move this into a method rather than inlining it.

Reply:
Do you want just the auto rect = ... moved into a function, or the whole lambda function?

Reply:
The entire lambda expression. The slot should be something similar to:

connect(..., ..., [=] (auto i) { on_scrollbar_value_changed(); });